main/main.go

```diff
@@ -246,19 +333,42 @@ func cleanUpAndSetup() error {
 	return nil
 }
 
-func generateDataFiles() error {
-	fmt.Printf("GenerateDataFiles routine started\n")
-	defer fmt.Printf("GenerateDataFiles routine completed\n")
+func (difftool *xdcrDiffTool) createFilterIfNecessary() error {
+	var ok bool
+	var expr string
+	if expr, ok = difftool.specifiedSpec.Settings.Values[metadata.FilterExpressionKey].(string); !ok {
```

Comment:
may have issues with older builds where Settings.Values does not exist

Reply:
XDCR difftool when compiled against MH build, will retrieve the old form, run upgrade code and populate the Values

Reply:
ok, if the assumption is that the difftool will always be compiled with latest MH builds.